### PR TITLE
JacobW scale cow sale price

### PR DESCRIPTION
**Pull request by JacobWoodward2000**
Added a check box that allows you to choose if you want cows at low health for sell for less, and updated the backend to be able to take this parameter and perform the correct calculation.

# Storybook:
Before:
* [CommonsForm](https://ucsb-cs156-s23.github.io/proj-happycows-s23-6pm-4/storybook/?path=/story/components-commons-commonsform--uncontrolled)
* [ManageCows](https://ucsb-cs156-s23.github.io/proj-happycows-s23-6pm-4/storybook/?path=/story/components-commons-managecows--uncontrolled)
* [CommonsTable](https://ucsb-cs156-s23.github.io/proj-happycows-s23-6pm-4/storybook/?path=/story/components-commons-commonstable--empty)

After:
* [CommonsForm](https://ucsb-cs156-s23.github.io/proj-happycows-s23-6pm-4/prs/25/storybook/?path=/story/components-commons-commonsform--uncontrolled)
* [ManageCows](https://ucsb-cs156-s23.github.io/proj-happycows-s23-6pm-4/prs/25/storybook/?path=/story/components-commons-managecows--uncontrolled) (text is hidden when it does not apply)
* [CommonsTable](https://ucsb-cs156-s23.github.io/proj-happycows-s23-6pm-4/prs/25/storybook/?path=/story/components-commons-commonstable--empty)

# Screenshots:
Before:
Screenshot 2023-05-31 181318
Screenshot 2023-05-30 143710
Screenshot 2023-05-31 182836
After:
Screenshot 2023-05-31 181333
Screenshot 2023-05-31 183408
Screenshot 2023-05-31 182851

### CODE REVIEW

File: frontend/src/main/components/Commons/ManageCows.js [hunk 1]
```diff
@@ -2,14 +2,16 @@ import React from "react";
 import { Card, Button, Row, Col } from "react-bootstrap";
 import cowHead from "./../../../assets/CowHead.png"; 
 
-// add parameters 
-const ManageCows = ({userCommons,commons, onBuy, onSell}) =>  {
+
+// add parameters
+const ManageCows = ({userCommons, commons, onBuy, onSell}) =>  {
     // update cowPrice from fixture
+    // Stryker disable ArithmeticOperator : line 24: not very important and unsure of how to test this
```
Comment (pconrad) [inline frontend/src/main/components/Commons/ManageCows.js:9]: Why is this here?   This keeps showing up in this team's PRs, and I'm skeptical.  I need to be convinced that this is necessary, especially since "line 24" seems to be misleading.

File: frontend/src/main/components/Commons/ManageCows.js [hunk 2]
```diff
@@ -2,14 +2,16 @@ import React from "react";
...
     return (
         <Card>
         <Card.Header as="h5">Manage Cows</Card.Header>
         <Card.Body>
-            {/* change $10 to info from fixture */}
+            {/* why does removing the optional chaining question mark turn the entire page white? */}
```
Comment (pconrad) [inline frontend/src/main/components/Commons/ManageCows.js:14]: Why does this comment keep showing up?

I've called it out in several code reviews.  Comments like this are fine while you are developing, but they need to be removed before code is ready to merge to main.

And I've answered this question before.

File: src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java [hunk 3]
```diff
@@ -45,742 +45,991 @@
...
+    @MockBean
+    UserCommonsRepository userCommonsRepository;
+
+    @MockBean
+    UserRepository userRepository;
+
+    @MockBean
+    CommonsRepository commonsRepository;
+
+    @Autowired
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/happiercows/controllers/UserCommonsControllerTests.java:57]: Is this just white space changes or are there real changes here to review?

This gets really frustrating for a reviewer.  Please try to avoid this.

**Review by pconrad — CHANGES_REQUESTED** [review]
Several issues.

File: frontend/src/main/components/Commons/ManageCows.js (hunk 2 above)
Comment (JacobWoodward2000) [reply frontend/src/main/components/Commons/ManageCows.js:14]: This is because it is outdated. I was waiting for BuySellMultipleCows to be approved before merging all the changes into other branches.
